src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/cursor/CursorActionRequestRestExecutorFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.executor.cursor;
+
+import com.amazon.opendistroforelasticsearch.sql.executor.Format;
+import org.elasticsearch.rest.RestRequest;
+
+public class CursorActionRequestRestExecutorFactory {
+    //TODO: add javadocs, see RestExecutor
+
+    public static CursorRestExecutor createExecutor(RestRequest request, String cursor, Format format) {
+        if (isCursorCloseRequest(request)) {
```

Comment:
what happen if it is an explain request

Reply:
As per plugin usage ,  `_explain` does make sense with `cursor`. Will handle this case.

Reply:
Handled this case. Added integration test for same.